src/pseudopeople/utilities.py

```diff
@@ -81,22 +76,20 @@ def get_index_to_noise(
 
     index_eligible_for_noise = dataset.get_non_empty_index(required_columns)
 
-    # As long as noise is relatively rare, it will be faster to randomly select cells to
-    # noise rather than generating a random draw for every item eligible
-    if isinstance(noise_level, float) and noise_level < 0.2:
-        rng = np.random.default_rng(
-            seed=get_hash(f"{dataset.randomness.seed}_get_index_to_noise_{additional_key}")
+    if isinstance(noise_level, float):
+        number_to_noise = dataset.randomness.binomial(
+            len(index_eligible_for_noise), p=noise_level
         )
-        number_to_noise = rng.binomial(len(index_eligible_for_noise), p=noise_level)
         to_noise_idx = pd.Index(
-            rng.choice(index_eligible_for_noise, size=number_to_noise, replace=False)
+            dataset.randomness.choice(
+                index_eligible_for_noise, size=number_to_noise, replace=False
+            )
         )
     else:
-        to_noise_idx = dataset.randomness.filter_for_probability(
-            index_eligible_for_noise,
-            probability=noise_level,
-            additional_key=additional_key,
-        )
+        # This is a copy paste of filter for probability
```

Comment:
why not just use `dataset.randomness.filter_for_probability` then?

Reply:
Thats not a method in numpy default_rng